### PR TITLE
Adds suggestion in the example autoalign/reproject to use `assume_spherical_screen()`

### LINKED ISSUE #5988: Suggest the use of `assume_spherical_screen()` in the autoalign/reproject examples
These two examples:
* Autoaligning: https://docs.sunpy.org/en/v3.1.5/generated/gallery/map_transformations/autoalign_aia_hmi.html
* Reprojecting: https://docs.sunpy.org/en/v3.1.5/generated/gallery/map_transformations/reprojection_align_aia_hmi.html

are silent about the fact that off-disk HMI data is being discarded.  The examples should note that, and additionally suggest that the user can use `Helioprojective.assume_spherical_screen()` to retain the off-disk HMI data.

### PR DESCRIPTION
**Pull request by davnish**
This PR adds a suggestion in these two examples:

- [Auto-Aligning AIA and HMI Data During Plotting](https://docs.sunpy.org/en/stable/generated/gallery/map_transformations/autoalign_aia_hmi.html#sphx-glr-generated-gallery-map-transformations-autoalign-aia-hmi-py)
- [Aligning AIA and HMI Data with Reproject](https://docs.sunpy.org/en/stable/generated/gallery/map_transformations/reprojection_align_aia_hmi.html#sphx-glr-generated-gallery-map-transformations-reprojection-align-aia-hmi-py)

This suggestion tells the user that Off-disks HMI data is not retained here, and to retain the data they can use `Helioprojective.assume_spherical_screen()` to retain the data. 

Merging this PR:
Closes #5988

### CODE REVIEW

**Comment by davnish** [comment]
pre-commit.ci autofix

File: examples/map_transformations/autoalign_aia_hmi.py [hunk 1]
```diff
@@ -45,10 +45,16 @@
...
-# the AIA map # instead of the HMI map.  We also need to manually set
+# the AIA map # instead of the HMI map. We also need to manually set
 # the plot limits because Matplotlib gets confused by the off-disk
-# parts of the image.  Note that the HMI image now has the same
+# parts of the image. We can see now that the HMI image now has the same
 # orientation as the AIA image.
+#
+# Note that off-disk HMI data is not retained here, we can use
+# :meth:`~sunpy.coordinates.Helioprojective.assume_spherical_screen` to
+# retain the off-disk HMI data. See,
```
Comment (ayshih) [inline examples/map_transformations/autoalign_aia_hmi.py:55]:
```suggestion
# retain the off-disk HMI data. See
```

File: examples/map_transformations/reprojection_align_aia_hmi.py [hunk 2]
```diff
@@ -55,6 +55,12 @@
 
 ######################################################################
 # Plot the images side by side.
+#
+# Note that off-disk HMI data is not retained here, we can use
+# :meth:`~sunpy.coordinates.Helioprojective.assume_spherical_screen` to
+# retain the off-disk HMI data. See,
```
Comment (ayshih) [inline examples/map_transformations/reprojection_align_aia_hmi.py:61]:
```suggestion
# retain the off-disk HMI data. See
```

**Review by ayshih — APPROVED** [review]
Some minor edits, otherwise good to go!

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks for the PR @davnish

**Pull request merged**